src/psyclone/transformations.py

```diff
@@ -119,6 +119,13 @@ def _validate(self, node_list):
                          if not isinstance(item, Schedule)]
             for item in flat_list:
                 if not isinstance(item, self.valid_node_types):
+                    # TODO: To the reviewer, Literals and References appear
+                    # everywhere now, should they be in all transformations
+                    # 'valid_node_types' or is it OK to skip them here. Also
```

Comment:
Yes, I think they should all appear in `valid_node_types`. As @rupertford pointed out though, we (I) probably need to re-think this functionality. Possibly we could reverse it and
have an `excluded_node_types` list but that's not for this issue.

Reply:
OK, I added them to all `valid_node_type`s, but I support the `excluded_nodes` as a better solution (and more readable to understand the current restrictions of each transformation).